### PR TITLE
Disabled cors policy by default; Reafactored fetch client

### PR DESCRIPTION
**Pull request by alexandrtovmach**

### CODE REVIEW

File: src/core/internal/fetch/fetchClient.ts [hunk 1]
```diff
@@ -1,7 +1,12 @@
 import { HttpClient } from '../..';
+import { RequestOptions } from 'http';
```
Comment (yevheniyJ) [inline src/core/internal/fetch/fetchClient.ts]: We need to avoid such imports from Node.js modules.

File: src/core/internal/fetch/fetchClient.ts [hunk 2]
```diff
@@ -27,24 +32,19 @@ export class FetchClient implements HttpClient {
...
+        const reqData = data || {};
 
+        await this.waitInQueue();
         return fetch(url, {
-            method: method,
-            headers: !!config ? config.headers : {},
-            body: body
+            headers: {
+                ...reqConfig.headers,
+                'Content-Type': 'application/json'
```
Comment (yevheniyJ) [inline src/core/internal/fetch/fetchClient.ts]: Will not work properly. 'Content-Type' header can be overridden.

File: src/core/internal/fetch/fetchClient.ts (hunk 1 above)
Comment (yevheniyJ) [reply src/core/internal/fetch/fetchClient.ts]: Proposal. Create interface near HttpClient interface with headers and cors flag and use it as a type of config in HttpClient.

Comment (alexandrtovmach) [reply src/core/internal/fetch/fetchClient.ts]: Ohhh... yeah, I forgot it

File: src/core/internal/fetch/fetchClient.ts (hunk 2 above)
Comment (alexandrtovmach) [reply src/core/internal/fetch/fetchClient.ts]: I thought that we use just json type, but okay will fix it

File: src/core/internal/fetch/fetchClient.ts [hunk 3]
```diff
@@ -27,24 +32,19 @@ export class FetchClient implements HttpClient {
...
-                config.headers = config.headers || {};
-                config.headers['Content-Type'] = 'application/json';
-            } else {
-                body = data;
-            }
-        }
-        await this.waitInQueue();
+    private async request(url: string, method: string, config?: IRequestConfig, data?: any) {
+        const reqConfig = config || {};
+        const reqData = data || {};
```
Comment (yevheniyJ) [inline src/core/internal/fetch/fetchClient.ts]: Also unsafe thing. For some calls we do not need to send any body in request. This code breaks this. Just keep in mind :
If we have body then attach in to request. If body is object then convert it by using `JSON.stringify` function and make sure that 'Content-Type' is 'application/json' (only if body != null and body is object). And no further changes in httpd config.

**Comment by andrii-bodnar** [comment]
Closed and reopened to trigger CI

**Pull request merged**